proxy/proxy.go

```diff
@@ -813,9 +812,21 @@ func (c *client) interceptSystemQuery(hdr *frame.Header, stmt interface{}) {
 		}
 	case *parser.UseStatement:
 		if _, err := c.proxy.maybeCreateSession(hdr.Version, s.Keyspace); err != nil {
-			c.send(hdr, &message.ServerError{ErrorMessage: "Proxy unable to create new session for keyspace"})
+			var cqlError *proxycore.CqlError
+			switch {
+			case errors.As(err, &cqlError):
+				errMsg := cqlError.Message
+				c.send(hdr, errMsg)
```

Comment:
Doesn't this also represent a ServerError?  CqlErrors come off of maybeCreateSession() for things like protocol violations: a bad handshake, a destination server complaining about auth, etc.  From the perspective of a client connected to cql-proxy don't those also represent ServerErrors?

Reply:
All I mean here is that this should be:

```
c.send(hdr, &message.ServerError{ErrorMessage: errMsg})
```

That way whatever subsequent processing the proxy applies to ServerErrors is also applied to this response... which seems correct.  And IIUC that should result in a meaningful message being delivered to the client as well, right?

This is a relatively minor point and shouldn't hold up a release or anything... I'm just wondering.